### PR TITLE
Update rule ID for 'Unnecessary call to Set.Contains'

### PR DESCRIPTION
**Pull request by mpidash**
Follow-up to #69179.

Another analyzer used CA1865, so this rule now uses CA1868.
Also fixed a diagnostic found during final testing of the PR for the analyzer.

### CODE REVIEW

File: eng/config/globalconfigs/Common.globalconfig
Comment (333fred) [inline eng/config/globalconfigs/Common.globalconfig:7]: In the future, we need to not check these in until the warning number is fixed. Another warning should not have come in and taken a number that had already effectively shipped to customers (us). I would prefer that we remove this entirely until the warning number is checked in.

Comment (mpidash) [reply eng/config/globalconfigs/Common.globalconfig:7]: I agree, that was my fault, sorry about that. I should have waited until the PR was merged and **then** opened the PR to fix the warnings.
The analyzer has been merged, so the warning number is fixed.

**Review by 333fred — APPROVED** [review]
LGTM (commit 2), provided tests are passing. @dotnet/roslyn-compiler for a second review.

**Review by jjonescz — APPROVED** [review]

**Pull request merged**
